src/s11/handlers/s11_DB_req_handler.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020, Infosys Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <string.h>
+#include <pthread.h>
+
+#include "err_codes.h"
+#include "s11_options.h"
+#include "ipc_api.h"
+#include "s11.h"
+#include "s11_config.h"
+#include "msgType.h"
+
+#include <gtpV2StackWrappers.h>
+/*Globals and externs*/
+extern int g_resp_fd;
+extern struct GtpV2Stack* gtpStack_gp;
+/*End : globals and externs*/
+
+int
+s11_DB_req_handler(MsgBuffer* message, GtpV2MessageHeader* hdr, uint32_t sgw_ip)
+{
+
+	struct db_req_Q_msg dbr_info = {0};
+
+	dbr_info.header.ue_idx = hdr->teid;
+	dbr_info.header.msg_type = delete_bearer_request;
+	dbr_info.seq_no = hdr->sequenceNumber;
+
+	DeleteBearerRequestMsgData msgData;
+	memset(&msgData, 0, sizeof(DeleteBearerRequestMsgData));
+
+	bool rc = GtpV2Stack_decodeMessage(gtpStack_gp, hdr, message, &msgData);
+	if(rc == false)
+	{
+			log_msg(LOG_ERROR, "s11_DB_req_handler: "
+					"Failed to decode Delete Bearer Request Msg %d\n",
+					hdr->teid);
+			return E_PARSING_FAILED;
+	}
+
+	if (msgData.linkedEpsBearerIdIePresent)
+	{
+	    dbr_info.linked_bearer_id = msgData.linkedEpsBearerId.epsBearerId;
+	}
+        else if (msgData.epsBearerIdsCount > 0)
+	{
+	    /*Rewriting msgData.epsBearerIdsCount as 1, since we support only one bearer ctxt in the Q msg struct currently*/
+	    msgData.epsBearerIdsCount = 1;
```

Comment:
I would suggest, let it go to Mme-app. Let Mme-app handle it and reject it. Since we are stopping create bearer it means delete bearer is not going to succeed .

Reply:
In that case, we need to store multiple bearer contexts in the cb_req_Q_msg , which requires an increase in DATA_BUF_SIZE (src/mme-app/ mmeThreads.h) and multiple bearer contexts in the CB_resp requires an increase in S11_IPC_MSG_BUF_LEN(src/s11/main.c).

Please let us know, if we can implement these changes.